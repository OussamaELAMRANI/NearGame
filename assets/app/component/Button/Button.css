.button {
    @apply inline-flex items-center px-4 py-2
    text-sm font-medium
    border border-transparent rounded-md shadow-sm
    bg-gray-70 hover:bg-gray-80 focus:ring-2 focus:ring-offset-2 focus:ring-gray-70
    ;
}

.button.primary {
    @apply text-white bg-coral-60 hover:bg-coral-70 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-coral-50;
}

.button.secondary {
    @apply text-white bg-tail-80 hover:bg-tail-90 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-tail-70;
}

.button.warning {
    @apply text-white bg-orange-60 hover:bg-brown-70 focus:ring-2 focus:ring-offset-2 focus:ring-orange-40;
}

.button.lg {
    @apply text-xl;
}

.button.xl {
    @apply text-2xl
}

.button.base {
    @apply text-base;
}

.button.sm {
    @apply text-xs;
}

.button[disabled]{
    @apply cursor-not-allowed disabled:opacity-50;
}

.button svg{
    @apply px-1.5 h-full
}
