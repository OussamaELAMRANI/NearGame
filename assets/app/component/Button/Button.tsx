import { FC } from 'react'
import * as PropTypes from 'prop-types'
import './Button.css'
import { ButtonProps } from '@/module/types'
import { trimString } from '@/helper/stringFormatter'

const Button: FC<ButtonProps> = ({ children, variant = '', size = 'base', ...props }) => {
  const classes = trimString(`${variant} ${size}`)

  return (<button className={`button ${classes}`} {...props}> {children} </button>)
}

Button.propTypes = {
  variant: PropTypes.oneOf(['primary', 'secondary'])
}
export default Button
