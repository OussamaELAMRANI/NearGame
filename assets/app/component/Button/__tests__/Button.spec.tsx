import { cleanup, render } from '@testing-library/react'
import { create } from 'react-test-renderer'
import Button from '../index'

describe('Button.tsx', () => {
  afterEach(cleanup)

  it('should render correctly !', () => {
    const elem = create(<Button />).toJSON()
    expect(elem).toMatchSnapshot()
  })

  it('should be basic button', function () {
    const btn = render(<Button>my button</Button>)
    expect(btn.container.firstChild).toHaveClass('button base')
  })
})
