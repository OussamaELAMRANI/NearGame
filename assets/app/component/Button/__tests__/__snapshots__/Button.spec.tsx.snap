// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button.tsx should render correctly ! 1`] = `
<button
  className="button base"
>
   
   
</button>
`;
